src/test/java/itest/SnapshotIT.java

```diff
@@ -0,0 +1,299 @@
+ *
+ * Subject to the condition set forth below, permission is hereby granted to any
+ * person obtaining a copy of this software, associated documentation and/or data
+ * (collectively the "Software"), free of charge and under any and all copyright
+ * rights in the Software, and any and all patent rights owned or freely
+ * licensable by each licensor hereunder covering either (i) the unmodified
+ * Software as contributed to or provided by such licensor, or (ii) the Larger
+ * Works (as defined below), to deal in both
+ *
+ * (a) the Software, and
+ * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
+ * one is included with the Software (each a "Larger Work" to which the Software
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package itest;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import io.cryostat.net.web.http.HttpMimeType;
+
+import io.vertx.core.MultiMap;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.handler.impl.HttpStatusException;
+import itest.bases.StandardSelfTest;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class SnapshotIT extends StandardSelfTest {
+    static final String TEST_RECORDING_NAME = "someRecording";
+    static final String V1_TARGET_REQ_URL =
+            String.format("/api/v1/targets%s", SELF_REFERENCE_TARGET_ID);
+    static final String V2_TARGET_REQ_URL =
+            String.format("/api/v2/targets%s", SELF_REFERENCE_TARGET_ID);
+    static final String SNAPSHOT_NAME_V1 = "snapshot-2";
+    static final String SNAPSHOT_NAME_V2 = "snapshot-4";
```

Comment:
When running all the integration tests with `mvn verify`, I can't guarantee that my snapshots will be the second and fourth recordings to be created. How else can I verify the snapshot names when I receive the response body?

Reply:
Or would it be enough to check that the recording name contains the string `snapshot` without checking the id?

Reply:
The success response from creating a snapshot should tell you what the name of the new snapshot is, so you could retain that value from the response and use it for cleanup or whatever else you need to do with it.

Reply:
As you point out, the actual name of that snapshot isn't precisely determinable in advance, so simply checking that the name fits an expected format (using a regex?) seems sufficient.